src/transformers/tokenization_utils.py

```diff
@@ -966,21 +977,25 @@ def convert_ids_to_tokens(
         Returns:
             `str` or `List[str]`: The decoded token(s).
         """
-        if isinstance(ids, int):
-            if ids in self._added_tokens_decoder:
-                return self._added_tokens_decoder[ids].content
-            else:
-                return self._convert_id_to_token(ids)
+        is_single_element = isinstance(ids, int)
+        if is_single_element:
+            ids = [ids]
+
         tokens = []
         for index in ids:
             index = int(index)
             if skip_special_tokens and index in self.all_special_ids:
                 continue
             if index in self._added_tokens_decoder:
                 tokens.append(self._added_tokens_decoder[index].content)
+            elif index >= self.vocab_size and index not in self.all_special_ids:
```

Comment:
The `added_tokens_decoder` should hold both special and non special. So if the index is ` >= self.vocab_size` then we should not have to check `self.all_special_ids`.